src/Illuminate/View/Compilers/Concerns/CompilesWhens.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Illuminate\View\Compilers\Concerns;
+
+trait CompilesWhens
+{
+    /**
+     * Compile the 'when' statement into valid PHP.
+     *
+     * @param  string  $expression
+     * @param  string  $output
+     * @param  bool  $escape
+     * @return string
+     */
+    protected function compileWhen($expression)
+    {
+        $expression = $expression == '()' ? "(false,'',false)" : $expression;
+
+        return "<?php when{$expression}; ?>";
```

Comment:
Also why is the empty expression needed? I would just remove that and it will throw an argument count error.

```suggestion
        return "<?php when{$expression}; ?>";
```

<img width="1451" alt="image" src="https://github.com/user-attachments/assets/54a56845-7732-49f5-be7f-f99d22f192ae">

Reply:
I've noticed a lot of other of these compiler expressions seem to do this, not 100% sure why.